pipeline.py

```diff
@@ -84,27 +99,69 @@ def _process(self, raw_text: str):
         """
         Processes each token and creates MorphToken class instance
         """
-        pass
+        pattern = re.compile('r[а-яА-Яa-zA-Z ёЁ]')
+        for symbol in raw_text:
+            if pattern.match(symbol) is None:
+                raw_text = raw_text.replace(symbol, '')
+
+        analyzed_text = Mystem().analyze(raw_text)
+        tokens = []
+        for token in analyzed_text:
+            if 'analysis' not in token:
+                continue
+            if not token['analysis']:
+                continue
+
+            morph_token = MorphologicalToken(token['text'])
+            morph_token.normalized_form = token['analysis'][0]('lex')
+            morph_token.tags_mystem = token['analysis'][0]('gr')
+            tokens.append(morph_token)
+        return tokens
 
 
 def validate_dataset(path_to_validate):
     """
     Validates folder with assets
     """
-    pass
 
+    if isinstance(path_to_validate, str):
+        path_to_validate = Path(path_to_validate)
 
-def main():
+    if not path_to_validate.exists():
+        raise FileNotFoundError
+
+    if not path_to_validate.is_dir():
+        raise NotADirectoryError
+
+    for file in list(path_to_validate.glob('*')):
+        if '.txt' in file.name:
```

Comment:
same here: instead of filtering like that, you can iterate over relevant files only by specifying filename pattern as an argument passed to glob

Reply:
OK, done